sample-apps/mqtt/string/consumer.js

```diff
@@ -0,0 +1,32 @@
+const mqtt = require('mqtt');
+const { VcdiffDecoder } = require('../../../lib');
+
+const options = {
+	keepalive: 30,
+	username: 'FIRST_HALF_OF_API_KEY',
+	password: 'SECOND_HALF_OF_API_KEY',
+	port: 8883
+};
+const client = mqtt.connect('mqtts:mqtt.ably.io', options);
+const channelName = 'sample-app-mqtt';
+const channelDecoder = new VcdiffDecoder();
+
+client.on('message', (_, payload) => {
+	let data = payload;
+
+	try {
+		if (VcdiffDecoder.isDelta(data)) {
+			data = channelDecoder.applyDelta(data).asUtf8String();
+		} else {
+			channelDecoder.setBase(data);
+		}
+		data = JSON.parse(data);

```

Comment:
### Pass 1: before I looked at your producer

Why involve JSON at all? If anything I would expect:

1. The delta application line to be `data = channelDecoder.applyDelta(data)` (i.e. drop the UTF8 step there)
2. This line to be `data = data.asUtf8String()` (doing the UTF8 decoding step for both cases)

Or, perhaps even better, start with a `let` for both `data` and `string` to make it clearer which one we're dealign with - ultimately calling `console.log('Received String: ' + string)` at the end.

### Pass 2: once I looked at your producer

ok, so now I get _why_ you have JSON getting involved here - it's because your producer generates strings that happen to be JSON formatted. But I think that's unnecessary noise. Surely just sending a simple string with an incrementing number on the end or something (albeit it'll have to be long enough to ensure deltas actually get generated, as I found when I was playing with this!) more clearly demonstrates what we're trying to show?

Reply:
Sending a JS object that gets serialized to JSON is a very common use case for the realtime users in my understanding. That's why I did the example this way - to reach larger potential audience.